recipes/LibriSpeech/SSL/train.py

```diff
@@ -0,0 +1,398 @@
+                outputs = self.compute_forward(batch, Stage.TRAIN)
+                objectives = self.compute_objectives(
+                    outputs, batch, Stage.TRAIN
+                )
+                loss = objectives["loss"]
+                if self.hparams.diversity_loss_weight == 0.0:
+                    backprop_loss = loss
+                else:
+                    backprop_loss = (
+                        loss
+                        + objectives["diversity_loss"] * self.hparams.diversity_loss_weight
+                        + objectives["latent_l2"] * self.hparams.latent_l2_penalty_weight
+                    )
+                (backprop_loss / self.grad_accumulation_factor).backward()
+                objectives["total_loss"] = backprop_loss.detach()
+                if should_step:
+                    self.optimizer.step()
+                    self.optimizer.zero_grad()
+                    self.optimizer_step += 1
+
+        if should_step:
+            if isinstance(
+                self.modules.model, DistributedDataParallel
+            ):
+                w2v_model = self.modules.model.module
+            else:
+                w2v_model = self.modules.model
+            if w2v_model.quantiser is not None:
+                w2v_model.quantiser.update_temp(
+                    self.optimizer_step
+                )
+
+            if (
+                hasattr(self.hparams, "log_interval")
+                and self.optimizer_step % self.hparams.log_interval
+                == 0
+            ):
+
+                log_dct = {
+                    k: (
+                        v.item()
+                        if isinstance(v, torch.Tensor)
+                        else v
+                    )
+                    for k, v in objectives.items()
+                }
+                current_lr = self.optimizer.param_groups[0]["lr"]
+                log_dct["lr"] = current_lr
+                log_dct["avg_loss"] = self.avg_train_loss
+
+                if hasattr(self, "time_last_log"):
+                    run_time_since_last_log = (
+                        time.time() - self.time_last_log
+                    )
+                    log_dct["stats/run_time"] = run_time_since_last_log
+                self.time_last_log = time.time()
+
+                log_str = f"Update: {self.optimizer_step} - Objectives: {log_dct}"
+                logger.info(log_str)
+
+                if self.hparams.use_wandb:
+                    run_on_main(
+                        wandb.log,
+                        kwargs={
+                            "data": log_dct,
+                            "step": self.optimizer_step,
+                        },
+                    )
+
+            self.hparams.lr_scheduler(
+                self.optimizer, self.optimizer_step
+            )
+
+        return objectives["loss"].detach()
+
+    def evaluate_batch(self, batch, stage):
```

Comment:
why is it redefined ? I don't see a difference with the standard process of Brain.

Reply:
I think this is something worth discussing, at the moment `evaluate_batch` is assumed to only return one value, but I would actually want to return both the loss and the accuracy (but this breaks because core.py will then try and use the return value to calculate the average loss)

you're right this change at the moment is pointless, when I was overriding `fit()` I had some changes here that I now removed.